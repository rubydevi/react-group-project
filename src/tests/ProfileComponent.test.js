import React from 'react';
import renderer from 'react-test-renderer';
import ProfileComponent from '../components/Profile';

// Mock the child components to avoid actual rendering
jest.mock('../components/JoinedMissions', () => 'JoinedMissions');
jest.mock('../components/ReservedRockets', () => 'ReservedRockets');
jest.mock('../components/ReservedDragons', () => 'ReservedDragons');

describe('ProfileComponent', () => {
  it('renders correctly', () => {
    const component = renderer.create(<ProfileComponent />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
