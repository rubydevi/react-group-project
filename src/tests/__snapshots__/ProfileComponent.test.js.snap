// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileComponent renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="profile-table-container"
  >
    <table
      className="profile-table"
    >
      <thead>
        <tr>
          <th>
            My Missions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <JoinedMissions />
          </td>
        </tr>
      </tbody>
    </table>
    <table
      className="profile-table"
    >
      <thead>
        <tr>
          <th>
            My Rockets
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <ReservedRockets />
          </td>
        </tr>
      </tbody>
    </table>
    <table
      className="profile-table"
    >
      <thead>
        <tr>
          <th>
            My Dragons
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <ReservedDragons />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
